src/util/Spotify.js

```diff
@@ -0,0 +1,90 @@
+let userToken; 
+let expTime;
+const clientID = '';
+const redirectURI = '';
+
+
+module.exports = {
+
+    getAccessToken: function() {
+        const pageLoc = window.location.href;
+
+        if (userToken) {
+            console.log(userToken)
+            return userToken;
+        };
+        
+        const userTokenMatch = pageLoc.match(/access_token=([^&]*)/);
+        const expTimeMatch = pageLoc.match(/expires_in=([^&]*)/);
+        
+        if (userTokenMatch && expTimeMatch) {
+            userToken = userTokenMatch[1];
+            expTime = Number(expTimeMatch[1]);
+            window.setTimeout(() => userToken = '', expTime * 1000);
+            window.history.pushState('Access Token', null, '/');
+            return userToken;
+        } else {
+            const accessUrl = 'https://accounts.spotify.com/authorize?client_id=' + clientID + '&response_type=token&scope=playlist-modify-public&redirect_uri=' + redirectURI;
+            window.location = accessUrl;
+        }
+    },
+
+    spotifySearch(term) {
+        const accessToken = module.exports.getAccessToken();
```

Comment:
Prefer not wrapping everything inside module.exports because it forces you to do `module.exports.getAccessToken()` instead of `getAccessToken()`